### ISSUE TITLE
Codebase refactoring, merging back

### ISSUE THREAD

**Issue by zxbodya** [SitePen/remap-istanbul#92]
Hi there,

Nice, to see project activity in recent time :)

Some time ago, 
after concluding that project is not actively maintained, 
I have forked it: https://github.com/zxbodya/istanbul-sourcemap-report

And now after seeing activity here, looking for options merging it back…

My goals, while doing fork were following:
- split to smaller, more maintainable files, refactor it, enforce eslint codestyle rules
- update to es6
- make better cli interface
- and of course fix some issues

I have made quite a lot of things there, mostly - to make it cleaner and easier to maintain:
- converted from amd to commonjs
- splited to smaller more maintainable modules
- updated to use es6 language features (with babel to build it for older nodejs versions when publishing)
- replaced jshint with eslint, fixed code-style issues
- changed testing framework to jasmine
- refactored some things
- and fixed issues I have

Also I have made some changes that likely can not be merged
1. changed CLI utility interface
2. removing grunt/gulp plugins, I was not using them so considered to not refactor them (maybe if merging we will need to revert them)

Generally, there is a lot of things that can be merged… But some of them are more about coding preferences, then about fixing actual problem.

I can create pull request with following changes: es6, commonjs, enforcing eslint rules, split to smaller modules, some refactoring, and babel build to support older nodejs versions. It is almost done already (need to rebase my commits on top of current master). Also it was done while keeping existing tests mostly not changed and passing. 

So what do you think about all this?

**Comment by dylans** [comment]
Thanks zxbodya!

Ok, some quick thoughts:

* We're not really likely to switch away from Intern, since we created and maintain Intern (and it has, in our opinion, a better set of features for testing than Jasmine)... I can appreciate that you chose Jasmine, perhaps out of familiarity or personal preference.
* ES6 is fine... if we were to do things today, we would either start there or probably use TypeScript
* Module format doesn't matter too much since this doesn't run in a browser, though I have heard of projects running Istanbul in a browser context, so maybe it does matter. We chose AMD at the time because it was the only format that easily runs in both browser and command-line without transpilation. We don't have a strong preference here.
* Better cli and fixing issues sounds great!
* Increased modularity is always good. We really did see remap-istanbul as a relatively small thing to quickly solve a need we had (I think the original version was created in a week or two at most), but it's grown a bit beyond that
* eslint and jscs have merged, so that seems reasonable to me

So yes, I think we would be happy to review and work with you on PRs. I think we're going to do a release today or early next week with the recently landed PRs, and then we can start on this?
